django/forms/models.py

```diff
@@ -1486,6 +1487,19 @@ def __init__(
         self.limit_choices_to = limit_choices_to  # limit the queryset later.
         self.to_field_name = to_field_name
 
+    def clean(self, value):
+        self.validate_non_null_characters(value)
+        return super().clean(value)
+
+    def validate_non_null_characters(self, value):
+        non_null_character_validator = ProhibitNullCharactersValidator()
+        if not hasattr(value, "__iter__"):
+            non_null_character_validator(value)
+            return
+
+        for item in value:
+            non_null_character_validator(item)
```

Comment:
```suggestion
        return ProhibitNullCharactersValidator()(value)
```
As `ModelChoiceField` doesn't know about `ModelMultipleChoiceField`, let's not handle the looping in here

Another question is on the name of this, maybe `validate_null_characters_prohibited` or `validate_no_null_characters` might be better (not the best at naming things)